app/controllers/collection_participants_controller.rb

```diff
@@ -14,37 +14,38 @@ def owners_required
     redirect_to collection_participants_path(@collection)
     false
   end
-  
+
   def no_participant
     flash[:error] = t('no_participant', :default => "Which participant did you want to work with?")
     redirect_to root_path
   end
-  
+
   def load_participant_and_collection
     if params[:collection_participant]
-      @participant = CollectionParticipant.find(params[:collection_participant][:id])
+      @participant = CollectionParticipant.where(id: params[:collection_participant][:id]).first
```

Comment:
Using `.find` in this line raises an ActiveRecord error if the participant isn't found, which then gives the user a 500. Since `no_participant` exists as a branch for this particular situation, I thought it'd make sense to just set a nil here and go down `no_participant` instead. Otherwise, there'd be no purpose in having `no_participant` and I'd recommend just getting rid of it. Open to either way, really.